### PR TITLE
Update WebAssembly debugging documentation, adjust templates

### PR DESCRIPTION
**Pull request by jeromelaban**
## PR Type

What kind of change does this PR introduce?
- Feature
- Documentation content changes

## What is the new behavior?

- Updated debugging documentation
- Updated templates to disable the linker in debug configuration by default
- Bump boostrapper to 3.0.0

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: doc/articles/debugging-wasm.md [hunk 1]
```diff
@@ -1,19 +1,38 @@
...
+- Using the browser's debugger
 
-- Make your WASM project the startup project (right-click **set as startup** in Solution Explorer)
-- Make sure you have the following lines defined in your project file which enable the Mono runtime debugger. Please ensure that `DEBUG` constant is defined and debug symbols are emitted and are of the type `portable`:
+## Using Visual Studio
+Here’s what you need to do to debug an Uno application in Visual Studio (2019 16.10+ or 2022 17.0 Preview 3.1+):
+
+- Install the latest [Uno Platform Visual Studio templates](https://marketplace.visualstudio.com/items?itemName=nventivecorp.uno-platform-addin)
+- Have Chrome or Edge (Chromium based)
+- In the Nuget Package manager, update `Uno.Wasm.Bootstrap` and `Uno.Wasm.Bootstrap.DevServer` 3.0.0 or later
```
Comment (Youssef1313) [inline doc/articles/debugging-wasm.md:12]:
```suggestion
- In the NuGet Package Manager, update `Uno.Wasm.Bootstrap` and `Uno.Wasm.Bootstrap.DevServer` 3.0.0 or later
```

**Review by Youssef1313 — COMMENTED** [review]
Will this work with the SamplesApp in `Uno.UI-Wasm-only.slnf`? Do we need any updates to csproj of the SamplesApp?

File: doc/articles/debugging-wasm.md [hunk 2]
```diff
@@ -1,19 +1,38 @@
...
-    <MonoRuntimeDebuggerEnabled>true</MonoRuntimeDebuggerEnabled>
-    <DefineConstants>$(DefineConstants);TRACE;DEBUG</DefineConstants>
-    <DebugType>portable</DebugType>
-    <DebugSymbols>true</DebugSymbols>
-</PropertyGroup>
-```
+You should now be able to set breakpoints or do step by step debugging of your code.
 
+### Tips for debugging in visual studio
+- Some debugger features may not have yet been implemented by the .NET and Visual Studio team. You can take a look the [dotnet/runtime](https://github.com/dotnet/runtime) repository for more details.
```
Comment (Youssef1313) [inline doc/articles/debugging-wasm.md:27]: Is there a link to a specific issue or document? Linking the repository isn't much helpful IMO.

File: doc/articles/debugging-wasm.md [hunk 3]
```diff
@@ -1,19 +1,38 @@
...
+- If the breaking do not hit, make sure that the `inspecturi` lines have been added to the `Properties/launchSettings.json` file.
+
+## Using the browser debugger
+
+To debug your application:
+- Make your WASM project the startup project (right-click **set as startup** in Solution Explorer)
+- In the Nuget Package manager, update `Uno.Wasm.Bootstrap` and `Uno.Wasm.Bootstrap.DevServer` 3.0.0 or later
```
Comment (Youssef1313) [inline doc/articles/debugging-wasm.md:34]:
```suggestion
- In the NuGet Package Manager, update `Uno.Wasm.Bootstrap` and `Uno.Wasm.Bootstrap.DevServer` to 3.0.0 or later
```

File: doc/articles/debugging-wasm.md (hunk 2 above)
Comment (jeromelaban) [reply doc/articles/debugging-wasm.md:27]: There's no specific link to features, but at least people know they can search there.

**Comment by jeromelaban** [comment]
> Will this work with the SamplesApp in `Uno.UI-Wasm-only.slnf`? Do we need any updates to csproj of the SamplesApp?

Yes it will. It'll be for another PR.

**Review by ajpinedam — APPROVED** [review]

**Pull request merged**